packages/opencensus-propagation-binaryformat/src/util.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      gRPC://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This util contains the functions for serializing and deserializing
+ * SpanContext using a binary format.
+ *
+ * See: https://github.com/census-instrumentation/opencensus-specs/blob/
+ *       master/encodings/BinaryEncoding.md
+ */
+
+import {SpanContext} from '@opencensus/core';
+import {decToHex, hexToDec} from 'hex2dec';
+
+const VERSION_ID = 0;
+const TRACE_ID_FIELD_ID = 0;
+const SPAN_ID_FIELD_ID = 1;
+const TRACE_OPTION_FIELD_ID = 2;
+
+// Sizes are number of bytes.
+const ID_SIZE = 1;
+const TRACE_ID_SIZE = 16;
+const SPAN_ID_SIZE = 8;
+const TRACE_OPTION_SIZE = 1;
+
+const FORMAT_LENGTH =
+    4 * ID_SIZE + TRACE_ID_SIZE + SPAN_ID_SIZE + TRACE_OPTION_SIZE;
+
+/**
+ * Serialize the given span context into a Buffer.
+ * @param spanContext The span context to serialize.
+ */
+export function serializeSpanContext(spanContext: SpanContext): Buffer {
+  /**
+   *  0           1           2
+   *  0 1 2345678901234567 8 90123456 7 8
+   * -------------------------------------
+   * | | |                | |        | | |
+   * -------------------------------------
+   *  ^ ^      ^           ^    ^     ^ ^
+   *  | |      |           |    |     | `-- options value (spanContext.options)
+   *  | |      |           |    |     `---- options field ID (2)
+   *  | |      |           |    `---------- spanID value (spanContext.spanID)
+   *  | |      |           `--------------- spanID field ID (1)
+   *  | |      `--------------------------- traceID value (spanContext.traceID)
+   *  | `---------------------------------- traceID field ID (0)
+   *  `------------------------------------ version (0)
+   */
+  const result = Buffer.alloc(FORMAT_LENGTH, 0);
```

Comment:
Is `Buffer` specific to Node.js? If so, would there be a way to do this that works in both browser and Node? (Maybe a typed array?)

Reply:
Done, changed to typed array instead of Buffer.

Reply:
Per our conversation offline, it sounds like a typed array will not work with the Node gRPC library, which expects a `Buffer` for binary headers. I think that's fine to keep this as a `Buffer`. Could you just add a comment that this uses a buffer for compatibility with the Node gRPC lib?

Reply:
Thanks @draffensperger, for the record -> A class for storing metadata has set method which requires ```key``` and ```value``` params. The ```value``` type must be a buffer if and only if the normalized key ends with '-bin' (in our case key name is "**grpc-trace-bin**")

Reply:
Done https://github.com/census-instrumentation/opencensus-node/pull/353/files?short_path=873f276#diff-54efdac81534fc9061a447d44bf49e5eR69 PTAL

Thanks @songy23 and @draffensperger for reviews.